tests/doc_index/elastic/v7/test_find.py

```diff
@@ -0,0 +1,271 @@
+        tens_two: NdArray = Field(dims=50)
+
+    store = ElasticDocumentIndex[FlatSchema]()
+
+    index_docs = [
+        FlatDoc(tens_one=np.random.rand(10), tens_two=np.random.rand(50))
+        for _ in range(10)
+    ]
+    store.index(index_docs)
+
+    query = index_docs[-1]
+
+    # find on tens_one
+    docs, scores = store.find(query, search_field='tens_one', limit=5)
+    assert len(docs) == 5
+    assert len(scores) == 5
+    assert docs[0].id == index_docs[-1].id
+    assert np.allclose(docs[0].tens_one, index_docs[-1].tens_one)
+    assert np.allclose(docs[0].tens_two, index_docs[-1].tens_two)
+
+    # find on tens_two
+    docs, scores = store.find(query, search_field='tens_two', limit=5)
+    assert len(docs) == 5
+    assert len(scores) == 5
+    assert docs[0].id == index_docs[-1].id
+    assert np.allclose(docs[0].tens_one, index_docs[-1].tens_one)
+    assert np.allclose(docs[0].tens_two, index_docs[-1].tens_two)
+
+
+def test_find_nested_schema():
+    class SimpleDoc(BaseDocument):
+        tens: NdArray[10]
+
+    class NestedDoc(BaseDocument):
+        d: SimpleDoc
+        tens: NdArray[10]
+
+    class DeepNestedDoc(BaseDocument):
+        d: NestedDoc
+        tens: NdArray = Field(dims=10)
+
+    store = ElasticDocumentIndex[DeepNestedDoc]()
+
+    index_docs = [
+        DeepNestedDoc(
+            d=NestedDoc(d=SimpleDoc(tens=np.random.rand(10)), tens=np.random.rand(10)),
+            tens=np.random.rand(10),
+        )
+        for _ in range(10)
+    ]
+    store.index(index_docs)
+
+    query = index_docs[-1]
+
+    # find on root level
+    docs, scores = store.find(query, search_field='tens', limit=5)
+    assert len(docs) == 5
+    assert len(scores) == 5
+    assert docs[0].id == index_docs[-1].id
+    assert np.allclose(docs[0].tens, index_docs[-1].tens)
+
+    # find on first nesting level
+    docs, scores = store.find(query, search_field='d__tens', limit=5)
+    assert len(docs) == 5
+    assert len(scores) == 5
+    assert docs[0].id == index_docs[-1].id
+    assert np.allclose(docs[0].d.tens, index_docs[-1].d.tens)
+
+    # find on second nesting level
+    docs, scores = store.find(query, search_field='d__d__tens', limit=5)
+    assert len(docs) == 5
+    assert len(scores) == 5
+    assert docs[0].id == index_docs[-1].id
+    assert np.allclose(docs[0].d.d.tens, index_docs[-1].d.d.tens)
+
+
+def test_find_batched():
+    store = ElasticDocumentIndex[SimpleDoc]()
+
+    index_docs = [SimpleDoc(tens=np.random.rand(10)) for _ in range(10)]
+    store.index(index_docs)
+
+    queries = index_docs[-2:]
+    docs_batched, scores_batched = store.find_batched(
+        queries, search_field='tens', limit=5
+    )
+
+    for docs, scores, query in zip(docs_batched, scores_batched, queries):
+        assert len(docs) == 5
+        assert len(scores) == 5
+        assert docs[0].id == query.id
+        assert np.allclose(docs[0].tens, query.tens)
+
+
+def test_filter():
```

Comment:
Can you add some test cases for `range` and `terms` filters? For example, selecting multiple filter terms for one field, and a range of integers/floats for another field.